File: lua/indenting.lua
-- vim: set nospell:
-- vim: set tabstop=2:
local M = {}

function M.GetLuaIndent()
  local indent = 99
  local contprev = vim.eval("getline(v:lnum-1)") -- Contenido de la línea anterior
  local indentprev = #(contprev:match("^%s*"))
  if type(contprev) == "string" and contprev:match("^%s*%- %[") then
    indent = indentprev + 6
  elseif type(contprev) == "string" and contprev:match("^%s*%- ") then
    indent = indentprev + 2
  elseif type(contprev) == "string" and contprev:match("^%* ") then
    indent = 2
  elseif type(contprev) == "string" and contprev:match("^%*%* ") then
    indent = 3
  elseif type(contprev) == "string" and contprev:match("^%*%*%* ") then
    indent = 4
  elseif type(contprev) == "string" and contprev:match("^%*%*%*%* ") then
    indent = 5
  elseif type(contprev) == "string" and contprev:match("^%*%*%*%*%* ") then
    indent = 6
  elseif type(contprev) == "string" and contprev:match("^%s.") then
    indent = indentprev
  else
    indent = indentprev
  end

  return indent
end


function M.GetPrevHeader()
  local linenum = vim.fn.line('.')
  local linecont = vim.fn.getline(linenum)
  while linenum > 0 do
    if linecont:match("^%* ") then
      previndent = 2
      break
    elseif linecont:match("^%*%* ") then
      previndent = 3
      break
    elseif linecont:match("^%*%*%* ") then
      previndent = 4
      break
    elseif linecont:match("^%*%*%*%* ") then
      previndent = 5
      break
    elseif linecont:match("^%*%*%*%*%* ") then
      previndent = 6
      break
    else
      linenum = linenum - 1 
    end
  end
end


function M.GetLuaIndentNvim(contline)
  contprev = vim.fn.getline(vim.v['lnum'] - 1)
  local indentprev = #(contprev:match("^%s*"))
  if contprev:match("^%* ") then
    previndent = 2
    return 2
  elseif type(contprev) == "string" and contprev:match("^%*%* ") then
    previndent = 3
    return 3
  elseif type(contprev) == "string" and contprev:match("^%*%*%* ") then
    previndent = 4
    return 4
  elseif type(contprev) == "string" and contprev:match("^%*%*%*%* ") then
    previndent = 5
    return 5
  elseif type(contprev) == "string" and contprev:match("^%*%*%*%*%* ") then
    previndent = 6
    return 6
  elseif type(contprev) == "string" and contprev:match("^%s*%- %[") then
    return indentprev + 6
  elseif type(contprev) == "string" and contprev:match("^%s*%- ") then
    return indentprev + 2
  elseif type(contprev) == "string" and contprev:match("^%s*%+ ") then
    return indentprev + 2
  elseif type(contprev) == "string" and contprev:match("^%s*.") then
    return indentprev
  else
    return previndent
  end
end
return M

File: python3/indenting.py
# vim: set foldmethod=indent:
import vim
import re
import utils.commandextend as comext

def RestoreHeaderLevel():
    global actualheader
    linenum = int(vim.eval("line('.')"))
    while linenum > 0:
        if vim.current.buffer[linenum].startswith('*'):
            if vim.current.buffer[linenum].startswith('* '):
                actualheader = 2
            elif vim.current.buffer[linenum].startswith('** '):
                actualheader = 3
            elif vim.current.buffer[linenum].startswith('*** '):
                actualheader = 4
            elif vim.current.buffer[linenum].startswith('**** '):
                actualheader = 5
            elif vim.current.buffer[linenum].startswith('***** '):
                actualheader = 6
            break
        else:
            linenum -= 1
    if linenum == 0:
        actualheader = 0


def CommandsHeads():
    command_list = [ 
                    ['OrgRestoreHeader',    'indenting', 'RestoreHeaderLevel()'], 
                    ]
    comext.OrgCommandBuffer( command_list )


def SetHeadersSettings():
    CommandsHeads()


def IndentVimWithPython():
    global actualheader
    global kindvlnum
    lcontent = vim.eval('getline(v:lnum-1)')
    # kindvlnum=vim.vvars['lnum']
    kindvlnum = int(vim.eval('v:lnum'))
    # print(type(lcontentprev))
    if lcontent.lstrip().startswith('- ['):
        return len(lcontent) - len(lcontent.lstrip()) + 5
    elif re.match(r'^[ \t]*$', lcontent):
        return actualheader
    elif lcontent.lstrip().startswith('-'):
        return len(lcontent) - len(lcontent.lstrip()) + 2
    elif lcontent.lstrip().startswith('* '):
        actualheader = 2
        return 2
    elif lcontent.startswith('** '):
        actualheader = 3
        return 3
    elif lcontent.startswith('*** '):
        actualheader = 4
        return 4
    elif lcontent.startswith('**** '):
        actualheader = 5
        return 5
    elif lcontent.startswith('***** '):
        actualheader = 6
        return 6
    elif re.match(r'^\s*+\d*[\.\)] ', lcontent):
        return int(vim.eval('indent(v:lnum-1)'))+lcontent.lstrip().index(' ') +1
    elif re.match(r'^\s*+[a-zA-Z][\.\)] ', lcontent):
        return int(vim.eval('indent(v:lnum-1)'))+lcontent.lstrip().index(' ') +1
    elif re.match(r'^\s*[a-zA-Z0-9:]', lcontent):
        return int(vim.eval('indent(getline(v:lnum-1))'))
    elif re.match(r'^[a-zA-Z0-9:#]', lcontent):
        return 0
    else:
        return actualheader
